scripts/experiments_images/ocsvm.jl

```diff
@@ -0,0 +1,112 @@
+using DrWatson
+@quickactivate
+using ArgParse
+using GenerativeAD
+using GenerativeAD.Models
+using BSON
+using StatsBase: fit!, predict, sample
+
+using MLDataPattern
+
+s = ArgParseSettings()
+@add_arg_table! s begin
+   "max_seed"
+		required = true
+		arg_type = Int
+		help = "seed"
+	"dataset"
+		required = true
+		arg_type = String
+		help = "dataset"
+	"anomaly_classes"
+		arg_type = Int
+		default = 10
+		help = "number of anomaly classes"
+end
+parsed_args = parse_args(ARGS, s)
+@unpack dataset, max_seed, anomaly_classes = parsed_args
+
+modelname = "OCSVM"
+
+function sample_params()
+	par_vec = (round.([10^x for x in -4:0.1:2],digits=5),["poly", "rbf", "sigmoid"])
+	argnames = (:gamma,:kernel)
+	return (;zip(argnames, map(x->sample(x, 1)[1], par_vec))...)
+end
+
+function joint_fit(models, data_splits)
+    info = []
+    for (model, data) in zip(models, data_splits)
+        push!(info, fit!(model, data))
+    end
+    return info
+end
+
+function joint_prediction(models, data)
+    joint_pred = Array{Float32}(undef, length(models), size(data,2))
+    for (i,model) in enumerate(models)
+        joint_pred[i,:] = predict(model, data)
+    end
+    return Flux.mean(joint_pred, dims=1)
+end
+
+function fit(data, parameters, n_models=10)
+    # construct model - constructor should only accept kwargs
+    models = [GenerativeAD.Models.OCSVM(;parameters...) for _ = 1:n_models]
+	
+	# sumbsample and fit train data
+	tr_data = data[1][1]
+    M,N = size(tr_data)
+    split_size = floor(N/n_models)
+    data_splits = [tr_data[:,Int(split_size*i+1):Int(split_size*(i+1))] for i = 0:n_models-1]
+
+	try
+		global info, fit_t, _, _, _ = @timed joint_fit(models, data_splits)
+	catch e
+		# return an empty array if fit fails so nothing is computed
+		return (fit_t = NaN,), [] 
+	end
+
+	# construct return information - put e.g. the model structure here for generative models
+	training_info = (
+		fit_t = fit_t,
+		model = nothing
+		)
+
+	# now return the different scoring functions
+	training_info, [(x->joint_predictiion(models, x), parameters)]
+end
+
+#_________________________________________________________________________________________________
+
+try_counter = 0
+max_tries = 10*max_seed
+
+while try_counter < max_tries
+	parameters = sample_params()
+
+	for seed in 1:max_seed
+		for i in 1:anomaly_classes
+			savepath = datadir("experiments/images/$(modelname)/$(dataset)/ac=$(i)/seed=$(seed)")
+
+			data = GenerativeAD.load_data(dataset, seed=seed, anomaly_class_ind=i)
+			
```

Comment:
Ignore my second comment, it is actually better to reshape the data here, e.g. write a nice function for it.

Reply:
i added function to utils and reshaping will take place here